Changes for running the docker tutorials successfully:

NicoletaOita opened the pull request:
- updated docker compose file for activemq to create the link between the gateway and the broker
- enabled some needed features at gateway startup (redis, http.proxy)

AdrianCozma commented on JMS/activemq/docker-compose.yml:
```diff
@@ -5,5 +5,9 @@ services:
+      - activemq:activemq_activemq
   activemq:
     build: broker
+    hostname: "kaazing.example.com"
```
why is this needed?

NicoletaOita replied:
I added it for the links, but removed it now, together with the links.

dpwspoon commented on enterprise-shield/AMQP/rabbitmq/docker-compose.yml:
```diff
@@ -5,7 +5,11 @@ services:
   internal.example.com: 
     build: internal
+    environment:
+      GATEWAY_OPTS: -Dfeature.http.proxy=true -Djava.naming.provider.url=FILE://kaazing-gateway/bin"
```
Why is the java.naming.provdier.url?  I wouldn't expect the .bindings file to be in bin.  And I don't think this would be needed for amqp service

NicoletaOita replied:
It wasn't needed, I forgot to remove it.

NicoletaOita commented:
@dpwspoon I've removed the links and the java naming provider url. I've also changed the gateway versions, added the necessary jars and commented the git clone part that returns an error for the javascript tutorials, such that you can see the exact changes that I've referred to in my [previous comment] (https://github.com/kaazing/QA-Roadmap/issues/35#issuecomment-238810771). Please let me know what you think.

dpwspoon commented:
This looks good to me.  I haven't / won't get a chance to run them.  @AdrianCozma please feel free to merge

dpwspoon commented:
Thanks @NicoletaOita

NicoletaOita commented:
@AdrianCozma I've replaced the old demos that used git with the already build ones from github, as @dpwspoon suggested [here] (https://github.com/kaazing/QA-Roadmap/issues/35#issuecomment-239034084). Please review.

dpwspoon commented on AMQP/rabbitmq/gateway/Dockerfile:
```diff
@@ -23,15 +22,10 @@ RUN set -ex \
-
-RUN git clone https://github.com/kaazing/javascript.client.tutorials ./web/javascript.client.tutorials
-RUN npm --prefix ./web/javascript.client.tutorials/amqp install ./web/javascript.client.tutorials/amqp 
+RUN curl -LOk  https://github.com/kaazing/javascript.client.tutorials/files/324464/websocket-jms-tutorial.zip
```
Nit: Best practices in docker is to combine this all of these into one "RUN" command to minimize layers, see how it is done on the left via &&.  Also ideally we would check the hash one it to be super secure

NicoletaOita replied:
@dpwspoon I've changed the Dockerfiles such that there is a single RUN command for downloading and unzipping the tutorials to the right locations.
Regarding the check to be secure, I assume you referred to the -k flag that I added, which doesn't check for the certificate, but if I don't use it, the tutorial is not downloaded.

dpwspoon commented:
Looks like progress to me :-)

AdrianCozma commented:
looks OK to me

dpwspoon commented:
Typically we build or create sha1 or .asc at build/release time.  I don’t think we generate them for the client tutorials, thus it isn’t needed for this PR.   Ideally we would already be generating them and you could use them.  There is a small possibility that the github API provides a way to get them.  Note: I suggested it as it is listed as best practice in security here: https://github.com/docker-library/official-images

NicoletaOita commented:
I've removed the sha1 check, please review.

AdrianCozma commented on JMS/activemq/docker-compose.yml:
```diff
@@ -6,4 +6,4 @@ services:
     hostname: "kaazing.example.com"
   activemq:
-    build: broker
+    image: webcenter/activemq:latest
```
I think we should use 5.10.0 instead of latest here

NicoletaOita replied:
@AdrianCozma If I replace the latest version with 5.10.0, I get the following error at gateway start:
```
gateway_1   | INFO  Started server successfully in 1.516 secs at 2016-08-18 11:37:02
gateway_1   | Exception in thread "ActiveMQ Connection Executor: tcp://activemq/172.18.0.2:61616@43328" java.lang.NullPointerException
gateway_1   |   at org.apache.activemq.ActiveMQConnection$5.run(ActiveMQConnection.java:2005)
gateway_1   |   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
gateway_1   |   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
gateway_1   |   at java.lang.Thread.run(Thread.java:745)
```

NicoletaOita replied:
I've filed a new [issue] (https://github.com/kaazing/tickets/issues/702) for the above mentioned behavior.
